openverse_catalog/dags/common/requester.py

```diff
@@ -5,6 +5,16 @@
 import requests
 
 
+# pytest_socket will not be available in production, so we must create a shim for
+# that specific exception so this code does not error out.
+try:
+    from pytest_socket import SocketConnectBlockedError
+except ImportError:
+
+    class SocketConnectBlockedError(Exception):
+        pass
```

Comment:
I followed the testing instructions and it behaved as expected -- slow pass as of c294e3e, and this one failure on stocksnap with the full branch:

```
pytest_socket.SocketConnectBlockedError: A test tried to use socket.socket.connect() with host "172.67.73.94" (allowed: "s3,postgres,172.19.0.3").

.local/lib/python3.10/site-packages/pytest_socket.py:186: SocketConnectBlockedError
```

So I believe that it works, but I'm not 100% sure _why_ it works. [StackOverflow](https://stackoverflow.com/questions/13886168/how-to-use-the-pass-statement) says a thing about "Deriving an exception class that does not add new behaviour". Is this how it works? Exceptions are classes, and when we say the "type" of the exception we literally mean the "type" as in the class name. So this is making sure that there is such a thing as a `SocketConnectBlockedError` and we always need to fully raise it as an error (not just throw it in the log, and move on), but we know for sure that it will only come up in testing, because ??? and even if it did come up in production, that's fine because it would just be like any abstract exception and raising it would offer some more useful info?

Reply:
The exception comes from the **pytest_socket** package that is only available as a [dev](https://github.com/WordPress/openverse-catalog/blob/main/requirements_dev.txt#L7) dependency, I believe that is why @AetherUnbound says it will not come up in production and also is the reason for this bit weird of setup :)

Reply:
Explained this a bit more in depth [here](https://github.com/WordPress/openverse-catalog/pull/629#discussion_r930513147), lemme know there if there's anything else confusing about this!